### PR TITLE
Test:  Egress to World fail if more than one connection of five failed.

### LINKED ISSUE #6102: CI: Retry external connectivity tests
- [ ] Require two consecutive failures to reach external endpoints before failing to make false positives less likely.

### PR DESCRIPTION
**Pull request by eloycoto**
Due the provider network issues, with this change we validate that no
more than one connection has been drop of five.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
fix  https://github.com/cilium/cilium/issues/6102

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6107)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/Policies.go [hunk 1]
```diff
@@ -1087,20 +1086,36 @@ var _ = Describe("RuntimePolicies", func() {
...
-			res = vm.ContainerExec(helpers.App1, helpers.CurlFail("-4 http://%s", googleHTTP))
-			ExpectWithOffset(2, res).Should(helpers.CMDSuccess(),
-				"not able to curl %s", googleHTTP)
+			By("Testing egress access to the world")
+			pingFailures := 0
+			curlFailures := 0
+
+			for i := 0; i < numberOfTries; i++ {
+				res := vm.ContainerExec(helpers.App1, helpers.Ping(googleDNS))
+				if !res.WasSuccessful() {
```
Comment (ianvernon) [inline test/runtime/Policies.go:1104]: Break out of the loop if ping and curl are successful. That way we won't have to connect multiple times unless we have to.

**Review by ianvernon — COMMENTED** [review]
LGTM, but please break out of the loop as requested.

File: test/runtime/Policies.go [hunk 2]
```diff
@@ -1087,20 +1086,36 @@ var _ = Describe("RuntimePolicies", func() {
...
+					pingFailures++
+				}
+				res = vm.ContainerExec(helpers.App1, helpers.CurlFail("-4 http://%s", googleHTTP))
+				if !res.WasSuccessful() {
+					curlFailures++
+				}
+			}
+
+			ExpectWithOffset(2, pingFailures).To(BeNumerically("=<", maxNumberOffFailures),
```
Comment (ianvernon) [inline test/runtime/Policies.go]: <=

File: test/runtime/Policies.go [hunk 3]
```diff
@@ -1087,20 +1086,36 @@ var _ = Describe("RuntimePolicies", func() {
...
+				"%d of %d  pings to %q failed", pingFailures, maxNumberOffFailures, googleDNS)
+
+			ExpectWithOffset(2, curlFailures).To(BeNumerically("=<", maxNumberOffFailures),
```
Comment (ianvernon) [inline test/runtime/Policies.go]: <=

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: test/runtime/Policies.go (hunk 1 above)
Comment (eloycoto) [reply test/runtime/Policies.go:1104]: Done!

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
Only the kafka test failed, merging

**Pull request merged**